src/Illuminate/Broadcasting/Broadcasters/AblyBroadcaster.php

```diff
@@ -22,108 +23,161 @@ class AblyBroadcaster extends Broadcaster
+            }
+            try {
+                $userChannelMetaData = parent::verifyUserCanAccessChannel($request, $normalizedChannelName);
+                if (is_array($userChannelMetaData) && array_key_exists('capability', $userChannelMetaData)) {
+                    $channelCapability = $userChannelMetaData['capability'];
+                    unset($userChannelMetaData['capability']);
+                }
+            } catch (\Exception $e) {
+                throw new AccessDeniedHttpException('Access denied, '.$this->stringify($channelName, $connectionId, $userId), $e);
+            }
+        }
 
-        $broadcastIdentifier = method_exists($user, 'getAuthIdentifierForBroadcasting')
-                    ? $user->getAuthIdentifierForBroadcasting()
-                    : $user->getAuthIdentifier();
+        try {
+            $signedToken = $this->getSignedToken($channelName, $token, $userId, $channelCapability);
+        } catch (\Exception $_) { // excluding exception to avoid exposing private key
+            throw new AccessDeniedHttpException('malformed token, '.$this->stringify($channelName, $connectionId, $userId));
+        }
 
-        $signature = $this->generateAblySignature(
-            $request->channel_name,
-            $request->socket_id,
-            $userData = array_filter([
-                'user_id' => (string) $broadcastIdentifier,
-                'user_info' => $result,
-            ])
-        );
+        $response = ['token' => $signedToken];
+        if (isset($userChannelMetaData) && is_array($userChannelMetaData) && count($userChannelMetaData) > 0) {
+            $response['info'] = $userChannelMetaData;
+        }
 
-        return [
-            'auth' => $this->getPublicToken().':'.$signature,
-            'channel_data' => json_encode($userData),
-        ];
+        return $response;
     }
 
     /**
-     * Generate the signature needed for Ably authentication headers.
+     * Return the valid authentication response.
      *
-     * @param  string  $channelName
-     * @param  string  $socketId
-     * @param  array|null  $userData
-     * @return string
+     * @param $request \Illuminate\Http\Request
+     * @param $result mixed
+     * @return mixed
      */
-    public function generateAblySignature($channelName, $socketId, $userData = null)
+    public function validAuthenticationResponse($request, $result)
     {
-        return hash_hmac(
-            'sha256',
-            sprintf('%s:%s%s', $socketId, $channelName, $userData ? ':'.json_encode($userData) : ''),
-            $this->getPrivateToken(),
-        );
+        return $result;
     }
 
     /**
      * Broadcast the given event.
      *
-     * @param  array  $channels
-     * @param  string  $event
-     * @param  array  $payload
+     * @param $channels array
+     * @param $event string
+     * @param $payload array
```

Comment:
```suggestion
     * @param  array  $channels
     * @param  string  $event
     * @param  array  $payload
```
First variable type and then the variable name

Reply:
Fixed!